saliency_metrics/metrics/sensitivity_n/sensn_result.py

```diff
@@ -0,0 +1,76 @@
+import os.path as osp
+from collections import defaultdict
+from typing import Dict, List
+
+import mmcv
+import numpy as np
+
+from ..serializable_result import SerializableResult
+
+
+class SensitivityNResult(SerializableResult):
+    """Helper class to record the Sensitivity-N training results.
+
+    Args:
+        summarized: if True, compute the mean and std of Pearson correlation coefficients for each ``n``
+
+    .. code-block:: python
+
+        from saliency_metrics.metrics SensitivityNResult
+
+        sensn_results = SensitivityNResult(summarized = True)
+        #Each single_result is a dictionary with "n" and the corresponding "correlation" for a image
+        id_result.add_single_result({"n": 1, "correlation": 0.1,})
+        id_result.add_single_result({"n": 1, "correlation": 0.2,})
+
+        #dump mean and std for each ``n``
+        sensn_results.dump("sensn_avg_result.json")
+
+        #The dump file should look like this:
+        #{"n": 1, "mean_correlation":0.15, "std_correlation": 0.05}
+    """
+
+    def __init__(self, summarized: bool = True) -> None:
+        self.summarized = summarized
+        self._corr_dict: defaultdict[str, List[str]] = defaultdict(list)
```

Comment:
The annotation should be `DefaultDict[str, List[float]]`. Please remember to import `DefaultDict` from `typing`.